dashboard/test/controllers/transfers_controller_test.rb

```diff
@@ -3,285 +3,228 @@
-      current_section_code: @word_section.code,
-      stay_enrolled_in_current_section: true
-    }
-    assert_not Follower.exists?(student_user: @word_user_1, section: @word_section)
+    @params[:stay_enrolled_in_current_section] = true
+    post :create, params: @params
+    assert_response 204
+    refute Follower.exists?(student_user: @word_student, section: @word_section)
   end
 
-  test "when the new section belongs to a different teacher, students should stay enrolled in the current section if stay_enrolled_in_current_section is true" do
-    new_section = create(:section, user: create(:teacher), login_type: 'word')
+  test "transferring to a new teacher errors without stay_enrolled_in_current_section" do
+    @params[:new_section_code] = @other_teacher_section.code
+    post :create, params: @params
+    assert_response 400
+    assert Follower.exists?(
+      student_user: @word_student,
+      section: @word_section
+    )
+    refute Follower.exists?(
+      student_user: @word_student,
+      section: @other_teacher_section
+    )
+  end
 
-    post :create, params: {
-      new_section_code: new_section.code,
-      student_ids: @word_user_1.id.to_s,
-      current_section_code: @word_section.code,
-      stay_enrolled_in_current_section: true
-    }
-    assert Follower.exists?(student_user: @word_user_1, section: @word_section)
+  test "when the new section belongs to a different teacher, students should stay enrolled in the current section if stay_enrolled_in_current_section is true" do
+    @params[:new_section_code] = @other_teacher_section.code
+    @params[:stay_enrolled_in_current_section] = true
+    post :create, params: @params
+    assert_response 204
+    assert Follower.exists?(student_user: @word_student, section: @word_section)
   end
 
   test "when the new section belongs to a different teacher, students should no longer be in the current section if stay_enrolled_in_current_section is false" do
-    new_section = create(:section, user: create(:teacher), login_type: 'word')
+    @params[:new_section_code] = @other_teacher_section.code
+    @params[:stay_enrolled_in_current_section] = false
 
-    post :create, params: {
-      new_section_code: new_section.code,
-      student_ids: @word_user_1.id.to_s,
-      current_section_code: @word_section.code,
-      stay_enrolled_in_current_section: false
-    }
-    assert_not Follower.exists?(student_user: @word_user_1, section: @word_section)
+    post :create, params: @params
+    assert_response 204
+    refute Follower.exists?(student_user: @word_student, section: @word_section)
   end
 
   test "transferring to the same section does nothing" do
-    post :create, params: {
-      new_section_code: @word_section.code,
-      student_ids: @word_user_1.id.to_s
-    }
-    assert Follower.exists?(student_user: @word_user_1, section: @word_section)
-  end
-
-  test "transferring a student to a section owned by a teacher the student "\
```

Comment:
This tests passes, because the `post` returns an error. While I investigate, I am simply commenting out the test.